test-d/jsonify.ts

```diff
@@ -122,3 +122,15 @@ class NonJsonWithInvalidToJSON {
 const nonJsonWithInvalidToJSON = new NonJsonWithInvalidToJSON();
 expectNotAssignable<JsonValue>(nonJsonWithInvalidToJSON);
 expectNotAssignable<JsonValue>(nonJsonWithInvalidToJSON.toJSON());
+
+// Special cases of Array with `undefined` member
+// `[undefined]` is not JSON because it contains non JSON value `undefined`
+// However `JSON.parse(JSON.stringify())` transforms array members of `undefined` to `null`
+expectNotAssignable<JsonValue>([undefined]);
+declare const parsedStringifiedArrayWithUndefined1: Jsonify<[undefined]>; // = JSON.parse(JSON.stringify([undefined]));
+expectType<null[]>(parsedStringifiedArrayWithUndefined1);
```

Comment:
Since it's expecting `null[]`, I think it should test `Jsonify<undefined[]>` for consistency.

Reply:
I am not sure what you mean.

I am verifying `Jsonify<undefined[]>` is the expected `null[]`.

`expectType<Jsonify<undefined[]>(parsedStringifiedArrayWithUndefined1);` would be true if the implementation of `Jsonify` was wrong.

Reply:
> I am verifying `Jsonify<undefined[]>` is the expected `null[]`.

I am saying that it should do that. However, it currently doesn't test `Jsonify<undefined[]>`, it tests `Jsonify<[undefined]>`.

Ultimately it isn't a big deal, I just didn't see any point in using a tuple.

Reply:
Ah - I get what you're saying now.  Thanks.  I will make that correction when I get back to this problem.

Reply:
Addressed in [68474d1](https://github.com/sindresorhus/type-fest/pull/310/commits/68474d1fd432c0ee16a94ab0a76bc65da179dba8)